src/ddprof_worker.cc

```diff
@@ -410,8 +414,10 @@ DDRes ddprof_pr_sample(DDProfContext &ctx, perf_event_sample *sample,
       // in lib mode we don't aggregate (protect to avoid link failures)
       int const i_export = ctx.worker_ctx.i_current_pprof;
       DDProfPProf *pprof = ctx.worker_ctx.pprof[i_export];
-      DDRES_CHECK_FWD(pprof_aggregate(&us->output, us->symbol_hdr, sample_val,
-                                      1, watcher, kSumPos, pprof));
+      const DDProfValuePack pack{static_cast<int64_t>(sample_val), 1,
+                                 ctx.params.timeline ? sample->time : 0};
```

Comment:
Currently timestamps have an unspecified origin, could this be an issue ?

Reply:
Indeed; I think this is a big problem for numerous reasons.

* Timeline currently has no means to extract the underlying clock source.  It is my belief that timeline should be able to comprehend this in order to determine whether two events on the same host can be ordered.  My primary interest in opening this PR was to add timestamps to fullhost in order to augment some work other teams are doing around sizing and cost analysis.  I think this will have enormous consequences for various products.
* Different events may have different underlying clock sources (in full generality) and there may be good reason to do so, in fact (e.g., if it's a high-frequency event or if we're bumping up against the kernel's event-taking overhead limit).
* As a matter of general principal, I think clocks are tricky and leaving the origin underspecified can only bring us trouble

At the same time, I wasn't so concerned about these topics that I felt the need to fix them in this PR--hopefully we can figure out a general strategy sometime, but it probably doesn't affect what we want to do next.  :)